web/js/precallController.js

```diff
@@ -65,6 +65,8 @@
     return new Promise(resolve => {
       if (window.routedFromStartMeeting) {
         publisherOptions.name = window.userName || document.querySelector(`${selector} input`).value.trim();
+        publisherOptions.publishAudio = document.querySelector(`${selector} #initialVideoSwitch label`).textContent === 'on';
+        publisherOptions.publishVideo = document.querySelector(`${selector} #initialAudioSwitch label`).textContent === 'on';
         return resolve({
```

Comment:
What is the source of the label being enabled? Might be a good idea to use source instead of  something that is using source

Reply:
Ive been struggling with this. This is the element:

<img width="380" alt="Vonage_Free_Conferencing" src="https://user-images.githubusercontent.com/9683228/97241749-c9ba5380-17d0-11eb-884e-6d51ecb2ff00.png">

Its a styled <a> tag. Instead of checking for the label value, i tried checking for the "activated" class to be present or not.
Whenever i do `.classList.value` on the element i get "activated" no matter if the class is there or not. Seems like its returning the class found when the doc loaded and its not updating dinamically.

I dont see a huge benefit on checknig the class vs checking the label value (which are the only two options i can think of), so imho id leave it as it is.
wdyt?

Reply:
yep agreed

Reply:
So, the thing we want to solve with this ticket, I think, is passing the video and audio selection (on/off) from the precallController used in the landingView along to the roomView. For example, serverMethods.js passes along the `routedFromStartMeeting` as an ejs template variable (to the client /room page), depending on whether we navigated from the landing page or not. Similarly, we should pass the audio and video settings (on/off) from the landing page (via ejs template variables) if we arrive at /room from a POST request from the landing page.

Reply:
I agree, I just wanted to stay consistent with the previous methodology of doing this. If you check line 67 we are doing the same for the username. Its directly grabbing it from the input field. If routedFromStartMeeting window.username is not defined. So it grabs it using the queryselector, analogous to what im doing here